src/components/Layout/Dropdown.astro

```diff
@@ -0,0 +1,137 @@
+---
+interface Props {
+  inlineMenu?: boolean;
+  label?: string;
+  class?: string;
+  trigger?: "click" | "hover";
+}
+
+// generate unique ids for each component so that we can connect the button to the dropdown without requiring consumer to provide an id
+// https://github.com/MicroWebStacks/astro-examples/blob/c278c3fc597bce013d627025a0c93758d72cc433/05_client-uid-counters/src/libs/utils.js#L8
+function uid() {
+  let date = (Date.now()).toString();
+  const sub = date.substring(date.length-6,date.length-1);
+  return sub+"_"+Math.floor(Math.random() * 10000)
+}
+
+let id = uid();
+let { inlineMenu, label, class: className, trigger } = Astro.props; 
+---
+
+<div
+  data-trigger={trigger ?? "click"}
+  class:list={["dropdown", "js-dropdown", { "dropdown-inline js-dropdown-inline": inlineMenu }, className]}
+> 
+  <button 
```

Comment:
Should this be a label because of that Safari bug? https://bugs.webkit.org/show_bug.cgi?id=22261